packages/core/contracts/financial-templates/optimistic-distributor/OptimisticDistributor.sol

```diff
@@ -0,0 +1,420 @@
+                customAncillaryData: customAncillaryData
+            });
+        uint256 rewardIndex = nextCreatedReward;
+        rewards[rewardIndex] = reward;
+        emit RewardCreated(
+            reward.sponsor,
+            reward.rewardToken,
+            rewardIndex,
+            reward.maximumRewardAmount,
+            reward.earliestProposalTimestamp,
+            reward.optimisticOracleProposerBond,
+            reward.optimisticOracleLivenessTime,
+            reward.priceIdentifier,
+            reward.customAncillaryData
+        );
+
+        // Bump nextCreatedReward index.
+        nextCreatedReward++;
+    }
+
+    /**
+     * @notice Allows anyone to deposit additional rewards for distribution before `earliestProposalTimestamp`.
+     * @dev The caller must approve this contract to transfer `additionalRewardAmount` amount of `rewardToken`.
+     * @param rewardIndex Index for identifying existing rewards object that should receive additional funding.
+     * @param additionalRewardAmount Additional reward amount that the sponsor is posting for distribution.
+     */
+    function increaseReward(uint256 rewardIndex, uint256 additionalRewardAmount) external nonReentrant() {
+        require(rewards[rewardIndex].sponsor != address(0), "Invalid rewardIndex");
+        require(getCurrentTime() < rewards[rewardIndex].earliestProposalTimestamp, "Funding period ended");
+
+        // Pull additional rewards from the sponsor.
+        rewards[rewardIndex].rewardToken.safeTransferFrom(msg.sender, address(this), additionalRewardAmount);
+
+        // Update maximumRewardAmount and log new amount.
+        rewards[rewardIndex].maximumRewardAmount += additionalRewardAmount;
+        emit RewardIncreased(rewardIndex, rewards[rewardIndex].maximumRewardAmount);
+    }
+
+    /********************************************
+     *          DISTRIBUTION FUNCTIONS          *
+     ********************************************/
+
+    /**
+     * @notice Allows any caller to propose distribution for funded reward starting from `earliestProposalTimestamp`.
+     * @dev The caller must approve this contract to transfer `optimisticOracleProposerBond` + final fee amount
+     * of `bondToken`.
+     * @param rewardIndex Index for identifying existing rewards object that should be proposed for distribution.
+     * @param merkleRoot Merkle root describing allocation of proposed rewards distribution.
+     * @param ipfsHash Hash of IPFS object, conveniently stored for clients to verify proposed distribution.
+     */
+    function proposeDistribution(
+        uint256 rewardIndex,
+        bytes32 merkleRoot,
+        string calldata ipfsHash
+    ) external nonReentrant() {
+        Reward memory reward = rewards[rewardIndex];
+        require(reward.sponsor != address(0), "Invalid rewardIndex");
+
+        uint256 timestamp = getCurrentTime();
+        require(timestamp >= reward.earliestProposalTimestamp, "No proposals in funding period");
```

Comment:
It is still funding period till `earliestProposalTimestamp`, so meaning no proposals are allowed yet.
To avoid confusion, changed to "Cannot propose in funding period" in 42be17adb2cec82c83ff2a93541d2d928e48e976